compiler/src/dmd/target.d

```diff
@@ -1136,6 +1136,15 @@ extern (C++) struct Target
     {
         return (os & Target.OS.Posix) != 0;
     }
+
+    /*********************
+     * Returns:
+     *  alignment of the stack
+     */
+    extern (C++) uint stackAlign()
+    {
+        return isXmmSupported() ? 16 : (is64bit ? 8 : 4);
```

Comment:
And I expect this to be wrong too. As I understand, minimum should always be 16 for SSE compat. Then AVX should be maximum 32, and AVX512F maximum 64.

Reply:
It's correct for what dmd generates. I don't know what gdc and ldc do with this, so please add what you need.

The alignment of the stack upon function entry is really 64 for AVX512F?

Reply:
> The alignment of the stack upon function entry is really 64 for AVX512F?

If the function handles a 512-bit vector, then I expect yes. Otherwise it's expected to be minimum 128bit on x86/x64. I will see if I can bring up some assembly examples from godbolt later.

Reply:
@WalterBright Here's an AVX512 example with GDC: https://d.godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1AB9U%2BlJL6yAngGVG6AMKpaAVxYMJAZlIOAMngMmABy7gBGmMQgAOykAA6oCoS2DM5uHt7xickCAUGhLBFRsZaY1ilCBEzEBGnunlw%2BpeUCldUEeSHhkTEWVTV1GY197Z0FRTEAlBaorsTI7Bx4LAk1ANRoxJgAdEks6ACkXgBCBxoAgmfnYnhMCmuBBFwAbGtHACJrpgBuZUTEEEeBwArMcXiD3pMjqcLlc4sQmMAWEw1oCGLRAphSGsqGIlJMHoIXoTkGinq9vtUobDojDzmsGWtKcQ3gAmU5eT5eaFXRlM6prABUHzWrJ5Fz5WwIcwY/OI4su0XeVyumFUBEisu%2BqDw6DWriUZOJqihJxVF21urWyMCEGpirpfMexrenLWXGiCr5qldn0CpJNXsZBswEEDZpp7w401onGBvE8HC0pFQnE%2BClm80wbK8PFIBE00emCEwTCwUTtpAA1iBWdFttEAJwADkbjQ0ki8kmizdZz2B%2Bk4kgThZTnF4ChAGnzhemcFgSGAyGQ6HccXIlDL6C2CgUAHk4jY7NxeGJaAej4m82FaKhkFX94eUpw82gWMiGOhjPDMDR1JW3ziOhImCVhFmAdBkHdVltlZXgsCqOhH0vF9eEwfB/l1PQaHoZg2GMBRXCoP9KGmfhBBEMR2CkGRBEUFR1CTXg8DiJh3jXUddB8DEggYVAL2fE9SDw9hKwRAB3ATjzzGpaHYlZR2aDV0GUO5J0rJTnwgBxBgaXxPzGbooi4LIkkvXS9ASMyUkMwoehMzTWn6WoXHqPRHIYNoalsiYHOciy/NGTFxns6YFAxFgpKvCcqgIZDBLzAgtkweS4lHJLMEwLDp1cXUotQ0htTieLpOjQcOHjYTR1TDgAHFHEcNYABVkoAegAJSavw1gANTwTBxMiNYIKgrgYNZVFcEIEgcxMtZnBWYCWQOVkvC8SZeALJjJmLUty1I6sJBgrhgWiLgNEbFtomiYFW0bbihxHJixw4CcpxnbbSHnRAUFQRb6DICgIEApaQBGsbWVILBvjwBY%2BoGg9GAKnCNWIdSwlHMJAmqABPAq3zYQQ93RPHnqwD9gHEMm8C2cpfknZ61TKVwNQKx5MFjZ6MTCBFiBx5wsHS4hllQsiDGABR4ckuIkaE8jhFEcQaPl%2Bi1E4kBuMMEwzH0PAwknSBplQJ8BAZgBaKg%2BLN7dXEEZZszNvc1jN5F5gQD5iDvKtMAIAxvYnDm/i0nTXKGfT0B8%2BzTJyVJQ70qyY8j4yLEDlpPP8uP3NTy8vI6YKjPcjP0j0hRnKTiQwszBYK/KyrE2TGrhsg6DYMmjCZpW3MNo%2BrQdtIEsyx6Ssay8YFtg7LxG2iZ5G2eZtp7rZ5yuHKrnpqt7py23uvpgH7ZgIOIWY3YG/qAgHQLYTgRpbuDfHb4X0Gw2RKKV6QVaUNXnq40hxIRNKTxjHGJ6DdOB7hZofAgaxUBUCbqNcaqIFpnyGp3dam1ZzTBrKyAcnMV7114OvCw70t5FnKrfPBL00HbWmL8NGz5JBAA%3D%3D%3D

Reply:
> @WalterBright Here's an AVX512 example with GDC:

Thanks, added SSE and AVX example too. https://d.godbolt.org/z/3MzncTGrj

Reply:
So, what it does is align the stack on function entry and then re-push the stack arguments. What this PR does is create an aligned section in the local stack frame for the variables needing bigger alignment values than the stack alignment. Either solution will work.

Reply:
If the use of this hook is versioned out in the front-end, I can just put its implementation as `assert(0)` on my end.